src/psyclone/psyir/transformations/arrayassignment2loops_trans.py

```diff
@@ -305,8 +306,11 @@ def validate(self, node, options=None):
         # We don't accept calls that are not guaranteed to be elemental
         for call in node.rhs.walk(Call):
             if isinstance(call, IntrinsicCall):
-                if call.intrinsic.is_inquiry:
-                    continue  # Inquiry intrinsic calls are fine
+                # Intrinsics that return scalars are fine.
+                if call.intrinsic in (IntrinsicCall.Intrinsic.LBOUND,
```

Comment:
These are not elemental (and are also fine because they always return a scalar - I now added the word "also" in the comment to make it more clear). Elemental intrinsics are handled just below with the generic `call.is_elemental`.